.github/workflows/push.yml

```diff
@@ -51,24 +51,18 @@ jobs:
         run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
         id: extract_branch
 
-  dispatch:
-    runs-on: ubuntu-latest
+  deploy:
     needs:
       - build_and_push
       - branch_name
-    steps:
-      - uses: actions/checkout@v2
-      - name: Extract branch name
-        shell: bash
-        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
-        id: extract_branch
-      - name: Repository Dispatch
-        uses: peter-evans/repository-dispatch@v1
-        with:
-          token: ${{ secrets.REPO_ACCESS_TOKEN }}
-          repository: hpi-schul-cloud/dof_app_deploy
-          event-type: dev-deploy
-          client-payload: '{"branch": "${{ steps.extract_branch.outputs.branch}}" }'
+    uses: hpi-schul-cloud/dof_app_deploy/.github/workflows/deploy.yml@main
+    with:
+      branch: ${{ needs.branch_name.outputs.branch }}
```

Comment:
an alternative way for push events to get branch name: 
```
ref: ${{ github.ref }} 
```
then set branch name from the argument if needed, in the remote workflow.
example https://github.com/hpi-schul-cloud/end-to-end-tests/blob/main/.github/workflows/e2e.yml#L14

then, in the remote workflow file,